frb_codegen/src/generator/dart/ty_boxed.rs

```diff
@@ -17,18 +17,40 @@ impl TypeDartGeneratorTrait for TypeBoxedGenerator<'_> {
                 self.ir.inner.safe_ident(),
             )
         });
+
+        let contains_opacity = self
+            .ir
+            .inner
+            .distinct_types(self.context.ir_file)
+            .iter()
+            .any(IrType::is_opaque);
```

Comment:
uber-nit: looks like `contains_opacity` is computed many times. What about extracting it to a function